server/server.js

```diff
@@ -129,6 +165,83 @@ app.post('/api/sign-in', async (req, res, next) => {
   }
 });
 
+// User can add to shopping cart
+app.post(
+  '/api/orderItems/:orderItemId',
+  authMiddleware,
+  async (req, res, next) => {
+    try {
+      const { orderId, productId, quantity } = req.body;
+      if (!orderId || !productId || !quantity)
+        throw new ClientError(
+          400,
+          'UserID, productID, and quantity are required fields'
+        );
+      const sql = `
+    insert into "orderItems" ("orderId", "productId", "quantity")
+    values ($1, $2, $3)
+    returning *;
+    `;
+      const params = [orderId, productId, quantity];
+      const result = await db.query(sql, params);
+      const [shoppingcart] = result.rows;
+      res.status(201).json(shoppingcart);
+    } catch (error) {
+      next(error);
+    }
+  }
+);
+
+// User can update shopping cart
+app.put(
+  '/api/orderItems/:orderItemId',
```

Comment:
This one is good. It lets the user update a specific, existing item in the cart.